frontend/lib/identification/scanner/qr_code_scanner.dart

```diff
@@ -0,0 +1,253 @@
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    isDone = false;
+    return Scaffold(
+      body: Column(
+        children: <Widget>[
+          Expanded(
+              flex: 4,
+              child: Scaffold(
+                appBar: AppBar(
+                  leading: new IconButton(
+                    icon: new Icon(
+                      Icons.arrow_back_ios,
+                      color: Theme.of(context).textTheme.bodyText1.color,
+                    ),
+                    onPressed: () => Navigator.of(context).pop(),
+                  ),
+                  title: Text("Ehrenamtskarte hinzufügen"),
+                ),
+                body: _buildQrView(context),
+              )),
+          Expanded(
+            flex: 1,
+            child: FittedBox(
+              fit: BoxFit.contain,
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                children: <Widget>[
+                  Container(
+                    margin: EdgeInsets.all(8),
+                    child: Text('Halten Sie die Kamera auf den QR Code.'),
+                  ),
+                  Row(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    crossAxisAlignment: CrossAxisAlignment.center,
+                    children: <Widget>[
+                      Container(
+                        margin: EdgeInsets.all(8),
+                        child: OutlineButton(
+                          onPressed: () {
+                            if (controller != null) {
+                              controller.toggleFlash();
+                              setState(() {
+                                flashState =
+                                    _isFlashOn(flashState) ? flashOff : flashOn;
+                              });
+                            }
+                          },
+                          child:
+                              Text(flashState, style: TextStyle(fontSize: 16)),
+                        ),
+                      ),
+                      Container(
+                        margin: EdgeInsets.all(8),
+                        child: OutlineButton(
+                          onPressed: () {
+                            if (controller != null) {
+                              controller.flipCamera();
+                              setState(() {
+                                cameraState = _isBackCamera(cameraState)
+                                    ? frontCamera
+                                    : backCamera;
+                              });
+                            }
+                          },
+                          child:
+                              Text(cameraState, style: TextStyle(fontSize: 16)),
+                        ),
+                      )
+                    ],
+                  ),
+                ],
+              ),
+            ),
+          )
+        ],
+      ),
+    );
+  }
```

Comment:
This is pretty big, we could think about extracting parts of that into a separate build function/widget.

Reply:
To be refactored later when revisited